workflows/smrnaseq.nf

```diff
@@ -0,0 +1,241 @@
+/*
+========================================================================================
+    VALIDATE INPUTS
+========================================================================================
+*/
+
+def summary_params = NfcoreSchema.paramsSummaryMap(workflow, params)
+
+// Validate input parameters
+WorkflowSmrnaseq.initialise(params, log)
+
+// Check input path parameters to see if they exist
+def checkPathParamList = [ params.input, params.multiqc_config ]
+for (param in checkPathParamList) { if (param) { file(param, checkIfExists: true) } }
+
+// Check mandatory parameters
+if (params.input) { ch_input = file(params.input) } else { exit 1, 'Input samplesheet not specified!' }
+
+// Check optional parameters
+if( !params.mirtrace_species ){
+    exit 1, "Reference species for miRTrace is not defined."
+}
+// Genome options
+bt_index_from_species = params.genome ? params.genomes[ params.genome ].bowtie ?: false : false
+bt_index = params.bt_indices ?: bt_index_from_species
```

Comment:
Maybe you could format the file this way? I think I had too much influence from @drpatelh :smile: 
```suggestion
bt_index_from_species = params.genome ? params.genomes[ params.genome ].bowtie ?: false : false
bt_index              = params.bt_indices ?: bt_index_from_species
```